src/common/vulkan/util.h

```diff
@@ -74,10 +74,141 @@ VkShaderModule CompileAndCreateFragmentShader(std::string_view source_code);
+template<> struct VkObjectTypeMap<VkShaderModule			> { using type = VkShaderModule				; static constexpr VkObjectType value = VK_OBJECT_TYPE_SHADER_MODULE; };
+template<> struct VkObjectTypeMap<VkPipelineCache			> { using type = VkPipelineCache			; static constexpr VkObjectType value = VK_OBJECT_TYPE_PIPELINE_CACHE; };
+template<> struct VkObjectTypeMap<VkPipelineLayout			> { using type = VkPipelineLayout			; static constexpr VkObjectType value = VK_OBJECT_TYPE_PIPELINE_LAYOUT; };
+template<> struct VkObjectTypeMap<VkRenderPass				> { using type = VkRenderPass				; static constexpr VkObjectType value = VK_OBJECT_TYPE_RENDER_PASS; };
+template<> struct VkObjectTypeMap<VkPipeline				> { using type = VkPipeline					; static constexpr VkObjectType value = VK_OBJECT_TYPE_PIPELINE; };
+template<> struct VkObjectTypeMap<VkDescriptorSetLayout		> { using type = VkDescriptorSetLayout		; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_SET_LAYOUT; };
+template<> struct VkObjectTypeMap<VkSampler					> { using type = VkSampler					; static constexpr VkObjectType value = VK_OBJECT_TYPE_SAMPLER; };
+template<> struct VkObjectTypeMap<VkDescriptorPool			> { using type = VkDescriptorPool			; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_POOL; };
+template<> struct VkObjectTypeMap<VkDescriptorSet			> { using type = VkDescriptorSet			; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_SET; };
+template<> struct VkObjectTypeMap<VkFramebuffer				> { using type = VkFramebuffer				; static constexpr VkObjectType value = VK_OBJECT_TYPE_FRAMEBUFFER; };
+template<> struct VkObjectTypeMap<VkCommandPool				> { using type = VkCommandPool				; static constexpr VkObjectType value = VK_OBJECT_TYPE_COMMAND_POOL; };
+template<> struct VkObjectTypeMap<VkDescriptorUpdateTemplate> { using type = VkDescriptorUpdateTemplate	; static constexpr VkObjectType value = VK_OBJECT_TYPE_DESCRIPTOR_UPDATE_TEMPLATE; };
+template<> struct VkObjectTypeMap<VkSurfaceKHR				> { using type = VkSurfaceKHR				; static constexpr VkObjectType value = VK_OBJECT_TYPE_SURFACE_KHR; };
+template<> struct VkObjectTypeMap<VkSwapchainKHR			> { using type = VkSwapchainKHR				; static constexpr VkObjectType value = VK_OBJECT_TYPE_SWAPCHAIN_KHR; };
+template<> struct VkObjectTypeMap<VkDebugUtilsMessengerEXT	> { using type = VkDebugUtilsMessengerEXT	; static constexpr VkObjectType value = VK_OBJECT_TYPE_DEBUG_UTILS_MESSENGER_EXT; };
+// clang-format on
+
+void SetObjectName(VkDevice device, void* object_handle, VkObjectType object_type, const char* object_name);
+
+template<typename T>
+inline void SetObjectName(VkDevice device, T object_handle, const char* object_name)
+{
+  SetObjectName(device, reinterpret_cast<void*>((typename VkObjectTypeMap<T>::type)object_handle),
+                VkObjectTypeMap<T>::value, object_name);
+}
+
+// Command buffer debug utils
+void BeginDebugScope(VkCommandBuffer commandBuffer, const char* scope_name,
```

Comment:
It might be better to just define these in the .h rather than the .cpp, with duplicate definitions (error prone if one side of the ifdef is updated but not the other).